gradle-jdks-setup-common/src/main/java/com/palantir/gradle/jdks/setup/common/CommandRunner.java

```diff
@@ -56,37 +46,44 @@ public static String run(List<String> commandArguments, Optional<File> directory
                 throw new RuntimeException(String.format(
                         "Failed to run command '%s'. "
                                 + "Failed with exit code %d.\nError output:\n\n%s\n\nStandard Output:\n\n%s",
-                        String.join(" ", commandArguments), exitCode, errorOutput, output));
+                        String.join(" ", processBuilder.command()), exitCode, errorOutput, output));
             }
             return output;
         } catch (IOException | InterruptedException e) {
             throw new RuntimeException(
-                    String.format("Failed to run command '%s'. ", String.join(" ", commandArguments)), e);
+                    String.format("Failed to run command '%s'. ", String.join(" ", processBuilder.command())), e);
         } catch (ExecutionException e) {
             throw new RuntimeException(
-                    String.format("Failed to get output for the command '%s'. ", String.join(" ", commandArguments)),
+                    String.format(
+                            "Failed to get output for the command '%s'. ", String.join(" ", processBuilder.command())),
                     e);
         } finally {
             executorService.shutdown();
         }
     }
 
-    public static void runWithInheritIO(List<String> commandArguments, File directory) {
-        ExecutorService executorService = Executors.newFixedThreadPool(2);
+    public static void runWithLogger(
+            ProcessBuilder processBuilder,
+            Function<InputStream, Void> stdOutputWriter,
```

Comment:
Could we use `Consumer<InputStream>` instead of `Function<InputStream, Void>`?

Reply:
Would avoid the `return null`s

Reply:
that is perfect, changed it [here](https://github.com/palantir/gradle-jdks/pull/382/commits/5ff102c5133aaf7d5c0ba96712eca0c9e75b77ec)